docs/source/guides/user/chapters/logging.rst

```diff
@@ -38,9 +39,26 @@ directory by running::
 
   $ avocado run --store-logging-stream STREAM[:LEVEL][,STREAM[:LEVEL][,...]
  
-This will produce ``$STREAM.$LEVEL`` files per each (unique) entry in the test
-results directory.
+This will produce ``$STREAM.$LEVEL`` files per each (unique) entry in two places.
+One in each test result directory where the stream is separated by test and one in
+job log directory where the stream will be gathered from all tests into one place.
+This might be useful when you don't use parallel run and you want to have logs
+from all tests in one place.
 
 .. note:: You have to specify separated logging streams. You can't use the
- built-in streams in this function.
+ built-in streams in this function. 
 
+
+Avocado log files
+-----------------
+
+Avocado use three main files for logging information on different levels.
+All of them are stored in ``{avocado_dir}/job-results/{job}/``:
+
+:job.log: Logs related to job run under avocado.job namespace.
+:full.log: All generated logs in one job run. It contains all logs from
+           avocado internal to logs from all tests. If you run your tests
+           in parallel the logs might be overlap  since the logs are not
+           sorted and they are stored at the time when they have been created.
```

Comment:
I assume this could be disabled by default? If not then I assume this can still be disabled for those of us with large logs that don't want duplication of the same data?

Reply:
But I can add and option for disabling this. @clebergnu what do you think?

Reply:
But if we have set the relevant stream storing switches won't `full.log` represent the same as separate log files, just put in one? In other words, no logging will be lost if users forget to enable it, it will just be separate logs.

@clebergnu Could you motivate this decision? IMO the `full.log` is next to useless when it comes to parallel runs where everything is jumbled together and I thought the future direction for avocado is towards parallelism and away from serial runs. Isn't `full.log` just a remnant of serial runs then? Why is it provided by default?

Reply:
My hunch is that users would complain louder about the lack of an all-in-one log file, than they would about the extra storage spent.  But, another characteristic of what I valued about a "full log file", would be to have logs from non-avocado sources.  See [here](https://github.com/avocado-framework/avocado/pull/5645#issuecomment-1520897968) what I mean by it.

One real-world use case is the [QEMU tests](https://gitlab.com/qemu-project/qemu/-/tree/master/tests/avocado), that happen to use loggers named after their own project, that is, [`qemu.*`](https://gitlab.com/qemu-project/qemu/-/blob/ac5f7bf8e208cd7893dbb1a9520559e569a4677c/python/qemu/machine/machine.py#L53).  Those logs are critical to understanding the outcome of tests.

We can require users to manually register all loggers they need, or have a default that will persist them by default in a `full.log` kind of file.  The same logic applies to per-test logging, so my point was not just about a job-wide full log file.

Reply:
While I completely agree and we have also lost critical messages in order to understand our own tests (in experimental branches) which is why I kept being an advocate of finding a nice and standard way to handle such libraries, I am not really sure such a full log file will be tractable with ten or more parallel workers. How do you intend on separating the external library qemu logs from one test from another?

Reply:
> We can require users to manually register all loggers they need, or have a default that will persist them by default in a full.log kind of file.

I also don't understand the need for users to manually register all loggers since IMO the default test behavior should always be verbose and easy to debug instead of requiring manual switches to do so. Especially if the larger verbosity could easily be grepped or filter out and doesn't mean much performance penalty. It would make more sense to me if users manually disable some stream from some namespace for being too verbose or even then just tweak its logging level instead of having to re-enable all the streams that are harder to re-generate than to filter out.

Reply:
> > We can require users to manually register all loggers they need, or have a default that will persist them by default in a full.log kind of file.
> 
> I also don't understand the need for users to manually register all loggers since IMO the default test behavior should always be verbose and easy to debug instead of requiring manual switches to do so. Especially if the larger verbosity could easily be grepped or filter out and doesn't mean much performance penalty. It would make more sense to me if users manually disable some stream from some namespace for being too verbose or even then just tweak its logging level instead of having to re-enable all the streams that are harder to re-generate than to filter out.

_"the default test behavior should always be verbose and easy to debug instead of requiring manual switches to do so"_ is actually the point I tried to make here.  Again using the QEMU tests, it would not be convenient for people running the tests, to have to create a command line or configuration file that would enable the logging of `qemu.*`, which is critical to understanding the test outcome, but it's not generated directly by the tests.

And your next point, _"It would make more sense to me if users manually disable some stream"_, is also what I've tried to communicate.  That's why I'm advocating for the root logger to have a handler that will persist everything by default in a `full.log` kind of file.

I guess the discussion is now mainly on whether a *job level* `full.log` file makes sense (assuming the test level does make sense).  I'll concede gladly to a compromise of having the test-level by default, and not having a job-level by default, because my concern is really with users being able to understand what *really* happened to a test, and not the order of events between (possible parallel) tests.

Reply:
> > I also don't understand the need for users to manually register all loggers since IMO the default test behavior should always be verbose and easy to debug instead of requiring manual switches to do so. Especially if the larger verbosity could easily be grepped or filter out and doesn't mean much performance penalty. It would make more sense to me if users manually disable some stream from some namespace for being too verbose or even then just tweak its logging level instead of having to re-enable all the streams that are harder to re-generate than to filter out.
> 
> _"the default test behavior should always be verbose and easy to debug instead of requiring manual switches to do so"_ is actually the point I tried to make here. Again using the QEMU tests, it would not be convenient for people running the tests, to have to create a command line or configuration file that would enable the logging of `qemu.*`, which is critical to understanding the test outcome, but it's not generated directly by the tests.

Exactly, but my point is somewhat bolder here: these `store-logging-stream` switches are not needed on the first place. Right now we take as default not to store these streams unless explicitly mentioned then try and cancel out our original default decision by introducing a full log file which might not have been needed on the first place. So I am on your opinion about what needs to be done but believe it could be done better. Because the same logging is just as critical for me, right now I am left with two choices: either try and shuffle a big and possibly scrambled full log file or manually restore each external library I need in order to understand the tests.

> I guess the discussion is now mainly on whether a job level full.log file makes sense (assuming the test level does make sense). I'll concede gladly to a compromise of having the test-level by default, and not having a job-level by default, because my concern is really with users being able to understand what really happened to a test, and not the order of events between (possible parallel) tests.

Indeed, this is perhaps a better partial solution of the above even though the nomenclature is not really ideal: `debug.log` sounds to me like it should contain everything or at least the names themselves don't clearly differentiate between `debug.log` and `full.log`. Perhaps `avocado.log` and `full.log`?